pkg/storage/iterator.go

```diff
@@ -87,35 +93,62 @@ func newBatchChunkIterator(ctx context.Context, chunks []*chunkenc.LazyChunk, ba
 			break
 		}
 	}
-
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I think this is going to leak because cancel never gets called:

```
WithCancel returns a copy of parent with a new Done channel. The returned context's Done channel is closed when the returned cancel function is called or when the parent context's Done channel is closed, whichever happens first.

Canceling this context releases resources associated with it, so code should call cancel as soon as the operations running in this Context complete. 
```

Reply:
yeah he forgot to copy my code entirely :)


```
func (it *batchChunkIterator) Close() error {
	it.cancel()
	if it.curr != nil {
		return it.curr.Close()
	}
	return nil
}
```

Reply:
you guys won't let me install any leaks....